src/main/java/com/pokegoapi/main/RequestHandler.java

```diff
@@ -129,8 +133,21 @@ public void sendServerRequests(ServerRequest... serverRequests) throws RemoteSer
 				lastAuth = responseEnvelop.getAuthTicket();
 			}
 
-			if (responseEnvelop.getStatusCode() == 102) {
-				throw new LoginFailedException();
+			if (responseEnvelop.getStatusCode() == 102 && GoogleLoginSecrets.refresh_token != null) {
```

Comment:
Maybe refactor the Login interface and implement this for Ptc as well, right now, by some bug in their checks of the token, those tokens are valid for a very long time, but I don't expect it to stay that way.

Then we can deprecate logging in with the token for Ptc as well and use some generic token refreshing code here (if `refreshToken` would be a part of the interface, the implementation for `PtcLogin` could currently be to return the old token).